### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by hay-k**
- Restructure documentation (move existing stuff around)
- Create a user guide
- Publish documentation online ([github pages](https://iqm-finland.github.io/cirq-on-iqm/index.html))

COMP-61

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -19,78 +21,28 @@ Currently Cirq on IQM can
...
-   $ tox -e docs
-   $ firefox build/sphinx/html/index.html
-
-
-Run the tests:
-
-.. code-block:: bash
-
-   $ tox
+Take a look at the `Jupyter Notebook <https://jupyter.org/>`_ with examples: ``examples/usage.ipynb``.
```
Comment (smite) [inline README.rst:45]:
```suggestion
Take a look at the `Jupyter notebook <https://jupyter.org/>`_ with examples: ``examples/usage.ipynb``.
```

File: README.rst [hunk 2]
```diff
@@ -1,15 +1,17 @@
-|CI badge|
+|CI badge| |Release badge|
 
 .. |CI badge| image:: https://github.com/iqm-finland/cirq-on-iqm/actions/workflows/ci.yml/badge.svg
+.. |Release badge| image:: https://img.shields.io/github/release/iqm-finland/cirq-on-iqm.svg
+
 
 Cirq on IQM
-###########
+===========
```
Comment (smite) [inline README.rst:8]:
```suggestion
###########
```

File: docs/index.rst [hunk 3]
```diff
@@ -1,12 +1,18 @@
+.. image:: images/IQM_Logo_RGB_frame.jpg
+    :width: 800
+
 Cirq on IQM
-###########
+===========
```
Comment (smite) [inline docs/index.rst:5]:
```suggestion
###########
```

File: README.rst (hunk 2 above)
Comment (smite) [reply README.rst:8]: It's the top-level heading.

File: docs/index.rst [hunk 4]
```diff
@@ -1,12 +1,18 @@
...
 
 :Release: |release|
 :Date: |today|
-
+:Source Code: `<https://github.com/iqm-finland/cirq-on-iqm>`_
 
 .. include:: ../README.rst
-   :start-line: 7
+    :start-line: 9
+    :end-before: Installation
```
Comment (smite) [inline docs/index.rst:13]: This is so specific it'll break when someone edits README :)
Maybe it's best just to copypaste the text here.

File: docs/user_guide.rst [hunk 5]
```diff
@@ -1,15 +1,207 @@
...
+
+You can also take a look at the function :func:`.extended_qasm_parser.circuit_from_qasm` which encompasses an
+OpenQASM 2.0 parser shipped with Cirq on IQM and supports an extended set of gate names.
+
+
+Running on a real quantum computer
+----------------------------------
+
+At this stage you can use your favorite Cirq simulator to simulate the circuits constructed above.
+In this subsection we demonstrate how to run them against a real IQM quantum chip.
```
Comment (olliahonen) [inline docs/user_guide.rst:156]:
```suggestion
In this subsection we demonstrate how to run them against a real IQM quantum computer.
```

File: docs/user_guide.rst [hunk 6]
```diff
@@ -1,15 +1,207 @@
...
+        measurements = result.measurements['mk']
+        print(measurements)
+
+
+More examples
+-------------
+
+More examples are available in the
+`examples directory <https://github.com/iqm-finland/cirq-on-iqm/tree/main/examples>`_
+of the Cirq on IQM github repository.
```
Comment (olliahonen) [inline docs/user_guide.rst:204]:
```suggestion
of the Cirq on IQM repository.
```

**Review by olliahonen — COMMENTED** [review]
A few minor comments

File: docs/index.rst (hunk 4 above)
Comment (hay-k) [reply docs/index.rst:13]: The best solution would be to have a separate `.rst` file containing the introduction and include it in both `README.rst` and `index.rst`, but unfortunately GitHub does not allow importing in the README. Nevertheless, duplicating the text seems to me less harmful than including like this )).

Comment (hay-k) [reply docs/index.rst]: Copied the text and removed the include statement

File: docs/user_guide.rst [hunk 7]
```diff
@@ -1,15 +1,258 @@
...
+a few additional gate types.
+
+
+Running on a real quantum computer
+----------------------------------
+
+.. note::
+
+   At the moment IQM does not provide a quantum computing service open to the general public,
+   so the instructions in this section are useful only to IQM customers.
```
Comment (olliahonen) [inline docs/user_guide.rst:205]:
```suggestion
   Please contact our `sales team <https://www.meetiqm.com/contact/>`_ to set up your access to an IQM quantum computer.
```

File: docs/user_guide.rst [hunk 8]
```diff
@@ -1,15 +1,258 @@
...
+:func:`.circuit_from_qasm` uses the OpenQASM 2.0 parser in :mod:`cirq.contrib.qasm_import`, extended by
+a few additional gate types.
+
+
+Running on a real quantum computer
+----------------------------------
+
+.. note::
+
+   At the moment IQM does not provide a quantum computing service open to the general public,
```
Comment (olliahonen) [inline docs/user_guide.rst:204]:
```suggestion
   At the moment IQM does not provide a quantum computing service open to the general public.
```

**Review by olliahonen — CHANGES_REQUESTED** [review]

File: docs/user_guide.rst [hunk 9]
```diff
@@ -1,15 +1,258 @@
...
+
+    qubit_mapping = {'Alice': 'QB1', 'Bob': 'QB3'}
+
+    with open(iqm_settings_path, 'r') as f:
+        sampler = IQMSampler(iqm_server_url, f.read(), adonis, qubit_mapping=qubit_mapping)
+
+    result = sampler.run(decomposed_circuit_2, repetitions=10)
+    print(result.measurements['m'])
+
+If you have circuit which already uses the device qubits, you don't need to specify
```
Comment (olliahonen) [inline docs/user_guide.rst:246]:
```suggestion
If you have a circuit which already uses the device qubits, you don't need to specify
```

File: docs/user_guide.rst [hunk 10]
```diff
@@ -1,15 +1,258 @@
...
+qubit mapping (as we did above for ``circuit_1``).
```
Comment (olliahonen) [inline docs/user_guide.rst:247]:
```suggestion
the qubit mapping (as we did above for ``circuit_1``).
```

File: src/cirq_iqm/iqm_remote.py [hunk 11]
```diff
@@ -62,8 +62,9 @@ class IQMSampler(cirq.work.Sampler):
 
     Args:
         url: Endpoint for accessing the server interface. Has to start with http or https.
-        settings: Settings for the quantum computer.
-        qubit_mapping: Dict that maps logical qubit names to physical qubit names. Must be injective.
+        settings: settings for the quantum computer
```
Comment (olliahonen) [inline src/cirq_iqm/iqm_remote.py:65]:
```suggestion
        Settings: settings for the quantum computer
```

File: src/cirq_iqm/iqm_remote.py [hunk 12]
```diff
@@ -62,8 +62,9 @@ class IQMSampler(cirq.work.Sampler):
...
+        device: quantum architecture to execute the circuit on
```
Comment (olliahonen) [inline src/cirq_iqm/iqm_remote.py:66]:
```suggestion
        device: Quantum architecture to execute the circuit on
```

File: src/cirq_iqm/iqm_remote.py [hunk 13]
```diff
@@ -62,8 +62,9 @@ class IQMSampler(cirq.work.Sampler):
...
+        qubit_mapping: injective dict that maps logical qubit names to physical qubit names
```
Comment (olliahonen) [inline src/cirq_iqm/iqm_remote.py:67]:
```suggestion
        qubit_mapping: Injective dictionary that maps logical qubit names to physical qubit names
```

**Review by olliahonen — APPROVED** [review]

**Pull request merged**